packages/snaps-controllers/src/snaps/SnapController.ts

```diff
@@ -2701,7 +2701,11 @@ export class SnapController extends BaseController<
    */
   async #assertSnapRpcRequestResult(handlerType: HandlerType, result: unknown) {
     switch (handlerType) {
-      case HandlerType.OnTransaction:
+      case HandlerType.OnTransaction: {
+        // Null is an allowed return value here
+        if (result === null) {
+          return;
+        }
         assertStruct(result, OnTransactionResponseStruct);
```

Comment:
Maybe add `null` as valid type to `OnTransactionResponseStruct`, and change it so the code below only runs if it's not null?